app/message-pin/server/pinMessage.js

```diff
@@ -42,7 +42,9 @@ Meteor.methods({
 			});
 		}
 
-		if (!hasPermission(Meteor.userId(), 'pin-message', message.rid)) {
+		const room = Meteor.call('canAccessRoom', message.rid, Meteor.userId());
+
+		if (room.t !== 'd' && !hasPermission(Meteor.userId(), 'pin-message', message.rid)) {
```

Comment:
This check will allow users to bypass the permission and pin messages on channels with the type different than `d`, this could happen if someone uses the pin api or try calling the method via console. Also the base of this idea is to nullify the permission on direct messages, which I don't think it is a good idea. My suggestion would be to create a new permission for pinning in direct messages.